components-core/src/main/java/com/adyen/checkout/components/base/Configuration.kt

```diff
@@ -0,0 +1,30 @@
+package com.adyen.checkout.components.base
+
+import android.os.Parcel
+import android.os.Parcelable
+import com.adyen.checkout.core.api.Environment
+import com.adyen.checkout.core.util.ParcelUtils
+import java.util.*
+
+abstract class Configuration protected constructor(
+    val shopperLocale: Locale,
+    val environment: Environment,
+    val clientKey: String
+) : Parcelable {
+
+    constructor(parcel: Parcel) : this(
+        parcel.readSerializable() as Locale,
+        parcel.readParcelable(Environment::class.java.classLoader)!!,
```

Comment:
I usually try to avoid having the `!!` operator but I don't know if there are even any better alternatives in this case 🤔

Reply:
i agree with not forcing most of the time but here I think we either need to go with nullable parameters or non-nulls with ```!!```. another idea could be we can use a fallback value, but i'm not sure if it's wise to use fallback values while creating an object from parcel. so i think out of those three this one is doable

Reply:
I agree, fallback is not good in this case because it hides a problem in the parcelization, throwing an exception is better, and `!!` will do just that. The other, maybe cleaner, way is to create extension methods that would read from the parcel and throw an exception if something is null when it shouldn't be, but we can do that in another PR if we really need it 👍